Add London Ruby User Group (LRUG)

h-lame opened the pull request:
The videos.yml file is generated by the lrug site and hosted at
https://lrug.org/rubyevents-video-playlist.yml so we've just copied it into the
rubyevents repo for this PR. Currently fixed at 2020 onwards as while we have
meetings on the site going back to 2006, all the links to recordings now go to
a private vimeo account due to the long-time venue / host going into
administration.

We would be interested in exploring something that allows us to ping ruby 
events when a change is published to our video playlist yml file to automatically
update / enter events.

We have sponsor data, but haven't added it, no have we added CFP or Schedule 
info as these seem (understandably) tailored to one-off events like conferences,
less so to meet-ups.  For example, our CFP never ends.

h-lame committed:
Add London Ruby User Group (LRUG)

The videos.yml file is generated by the lrug site and hosted at
https://lrug.org/rubyevents-video-playlist.yml so we've just copied it into the
rubyevents repo for this PR. Currently fixed at 2020 onwards as while we have
meetings on the site going back to 2006, all the links to recordings now go to
a private vimeo account due to the long-time venue / host going into
administration.

h-lame committed:
Add comments to the event images about sizes

I found it useful to know the dimensions when creating these images for LRUG,
but didn't see it documented elsewhere.

h-lame committed:
Provide scripts/import_event for reimporting a single org

Run as

    bundle exec rails runner scripts/import_event -- lrug

This will look through data/organisations.yml to find the org with the supplied
slug and run the reimport for that org. This is otherwise a copy and paste of
the organisation creation part of db/seeds.rb.

I found this useful while I was prepping the LRUG data, so I hope it's useful
for others.

h-lame committed:
Run prettier over the lrug videos.yml

This yml file was generated from lrug.org via a `to_yaml` call, but the default
output isn't in keeping with the default yaml prettier rules which is annoying.

marcoroth committed:
Set type to meetup and remove .png assets

marcoroth reviewed (approved):
Thank you so much for tackling this @h-lame, this is awesome! I'm just gonna mention this PR here for future reference: https://github.com/lrug/lrug.org/pull/406

I set the `type: "meetup"` which should remove the "Sponsors", "CFP" and "Schedule" Tabs for now, since we haven't really optimized those for Meetup Groups yet, as you noticed. I also removed the unused .png assets, since we only really use the .webp ones.

We are in the middle of trying to figure out how to automate a lot of things for RubyEvents, similar to what we do with RubyConferences.org.

Your `videos.yml` generator is awesome and is already a great start! I feel like as long as we don't have a good way of automating things in general it's probably fine for now the keep updating it manually and running prettier/the lint step as part of that workflow.

I also love the `scripts/import_event` script you wrote, thanks for that! I've been meaning to do that for quite some time since the seeding can take a bit to finish. 

A few nits about the `videos.yml` file:

* I think your `published_at` timestamp is our `announced_at` timestamp (when the meetup/talk was announced).
* Our `published_at` timestamp is being used for when the recording of the talk got published, if at all. So if there is no `published_at` timestamp the talk recording isn't available. I usually set both `announced_at` and `published_at` to `"TODO"` if they are not known.
* We usually have all talks/events in `videos.yml` in `date: asc` order, while yours is in `date: desc`. Is there an easy way to update this?

But otherwise, this looks good to go from my side, thanks for working on this! <3

h-lame committed:
Update lrug videos.yml file

Extracted from a branch version on the lrug/lrug.org repo to address feedback:

1. put things in oldest first order
2. sort out published_at vs announced_at

h-lame commented:
@marcoroth I think I've fixed the `videos.yml` in this latest commit.

Does raise two questions though:

1. I've set both `published_at` and `announced_at` for the container talk that represents the meeting to be the publish date of the meeting writeup but for the individual talks inside the meeting I've set `announced_at` to the publish date of the meeting and `published_at` to "TODO".  Does that work?
2. Our talk descriptions are markdown formatted and I've just copied that in directly to the videos.yml description field - do I need to do something to deal with that?  Strip it out?

I'm tracking any changes to the generation of the file over in https://github.com/lrug/lrug.org/pull/407

h-lame commented:
> Hey @h-lame, sorry for the delay on this. It's not forgotten! I will get back to this after my Rails World talk this Friday! 🙏🏼

Not a problem!  Good luck with the talk!

h-lame committed:
Update videos for LRUG

* Add videos to Aug 2025 meeting now we've published them
* Add Sept 2025 meeting with videos
* Add Oct LRUG meeting

ismasan commented:
Hi! Anything else needed to move this forward?

marcoroth commented:
Maybe we can get https://github.com/lrug/lrug.org/pull/407 merged first and then the `videos.yml` in this PR updated based on that?

marcoroth committed:
Update

marcoroth committed:
Merge branch 'main' into add-lrug

marcoroth committed:
Update

marcoroth committed:
Update import event script

marcoroth committed:
Lint

marcoroth committed:
Update videos.yml file

marcoroth committed:
Update announced_at

marcoroth commented:
Thanks @h-lame! 🙏🏼 

Finally got this merged! I opened https://github.com/lrug/lrug.org/pull/430 with the required changes so we can easily update and add future events as well.
